### PR TITLE
feat: project deploy commit diff

### PR DESCRIPTION
**Pull request by abeizn**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
enhance refdiff to support project. 
1. supprot new commits_diffs table.
2. adapt project params
3. add e2e test

### Does this close any open issues?
related to #3685 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/refdiff/e2e/raw_tables/cicd_pipeline_commits.csv [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
+github:GithubRun:1:1,2022-11-17 06:35:35.758,2022-11-17 06:35:35.758,"{""ConnectionId"":1,""Owner"":""apache"",""Repo"":""incubator-devlake-website""}",_raw_github_api_runs,1496,"",commit_sha1,main,github:GithubRepo:1:484251804,""
+github:GithubRun:1:2,2022-11-17 06:35:35.758,2022-11-17 06:35:35.758,"{""ConnectionId"":1,""Owner"":""apache"",""Repo"":""incubator-devlake-website""}",_raw_github_api_runs,1495,"",commit_sha2,main,github:GithubRepo:1:484251804,""
+github:GithubRun:1:3,2022-11-17 06:35:35.758,2022-11-17 06:35:35.758,"{""ConnectionId"":1,""Owner"":""apache"",""Repo"":""incubator-devlake-website""}",_raw_github_api_runs,1494,"",commit_sha3,fix-edit-url,github:GithubRepo:1:484251804,""
+github:GithubRun:1:4,2022-11-17 06:35:35.758,2022-11-17 06:35:35.758,"{""ConnectionId"":1,""Owner"":""apache"",""Repo"":""incubator-devlake-website""}",_raw_github_api_runs,1493,"",commit_sha4,main,github:GithubRepo:1:484251804,""
+github:GithubRun:1:5,2022-11-17 06:35:35.758,2022-11-17 06:35:35.758,"{""ConnectionId"":1,""Owner"":""apache"",""Repo"":""incubator-devlake-website""}",_raw_github_api_runs,1492,"",commit_sha5,main,github:GithubRepo:1:484251804,""
```
Comment (warren830) [inline plugins/refdiff/e2e/raw_tables/cicd_pipeline_commits.csv:6]: Please add a blank line

Comment (warren830) [reply plugins/refdiff/e2e/raw_tables/cicd_pipeline_commits.csv:6]: And please ignore irrelevant fields like raw_*

**Review by likyh — CHANGES_REQUESTED** [review]
Although I understand, I think the logic that convert `commitPairs` to `CommitDiff` should be extract to a func.

**Review by likyh — APPROVED** [review]

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**